File: src/test/java/dev/sanket/happynumber/SolutionTest.java
package dev.sanket.happynumber;

import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.ValueSource;

import static org.assertj.core.api.Assertions.assertThat;

class SolutionTest {

    private final Solution solution = new Solution();

    @ParameterizedTest
    @ValueSource(ints = {1, 19})
    public void shouldPass(int n) {
        assertThat(solution.isHappy(n)).isTrue();
    }

    @ParameterizedTest
    @ValueSource(ints = {2})
    public void shouldPassForNonHappyNumbers(int n) {
        assertThat(solution.isHappy(n)).isFalse();
    }
}

File: src/test/kotlin/dev/sanket/linkedlistcycle/SolutionTest.kt
package dev.sanket.linkedlistcycle

import assertk.assertThat
import assertk.assertions.isFalse
import dev.sanket.dev.sanket.linkedlistcycle.Solution
import org.junit.jupiter.api.Test

class SolutionTest {

    private val solution = Solution()

    @Test
    fun shouldReturnFalseWhenListIsNull() {
        val hasCycle = solution.hasCycle(null)

        assertThat(hasCycle).isFalse()
    }
}
